src/auth/mas.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Helper module used to manage MAS-related workloads."""
+
+import logging
+
+import ops
+from jinja2 import Environment, FileSystemLoader, select_autoescape
+
+from state.charm_state import SynapseConfig
+from state.mas import MASConfiguration
+
+MAS_TEMPLATE_FILE_NAME = "mas_config.yaml.j2"
+
+logger = logging.getLogger()
+
+
+# pylint: disable=too-few-public-methods
+class MasService:
```

Comment:
Well maybe we don't need a class, none of the instance attributes are used. A module with only `generate_mas_config` might be sufficient. Unless a later PR adds more to this module (I don't remember if there are)

Reply:
Make sense, initially this was a class because it had the logic to interact with juju secret, now that's in the state component I removed the class entirely.